snowpack/src/types.ts

```diff
@@ -224,10 +224,17 @@ export interface OptimizeOptions {
   target: 'es2020' | 'es2019' | 'es2018' | 'es2017';
 }
 
+export type HeadersTransformer = (
+  req: http.IncomingMessage,
+  proposed: http.OutgoingHttpHeaders | http.OutgoingHttpHeader[] | undefined
+) => http.OutgoingHttpHeaders | http.OutgoingHttpHeader[] | undefined;
+
 export interface RouteConfigObject {
-  src: string;
+  /** @default '.*' */
+  src?: string;
   dest: string | ((req: http.IncomingMessage, res: http.ServerResponse) => void) | undefined;
   upgrade: ((req: http.IncomingMessage, socket: net.Socket, head: Buffer) => void) | undefined;
+  headers?: HeadersTransformer;
```

Comment:
yup; considered naming it like `customizeHeaders`, but settled on matching the naming convention I saw. since there seems to be some support for `customizeHeaders` naming: happy to rename.

will add jsdoc to the parameter.